components/function-controller/pkg/apis/serverless/v1alpha1/function_types.go

```diff
@@ -5,6 +5,15 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+type SourceType string
+
+type Runtime string
```

Comment:
`// +kubebuilder:validation:Enum=...` marker could be added to ensure validation on the API level.

Reply:
yeah, same for sourcetype